capella2polarion/elements/capella_object_renderer.py

```diff
@@ -234,23 +237,40 @@ def render_jinja_template(
         env = self._get_jinja_env(str(template_folder))
         template = env.get_template(str(template_path))
         rendered_jinja = template.render(
-            object=capella_element,
+            object=converter_data.capella_element,
             model=self.model,
-            work_item=work_item,
+            work_item=converter_data.work_item,
+            config=converter_data.type_config,
             **(render_params or {}),
         )
-        return self.sanitize_text(capella_element, rendered_jinja, errors)
+        return self.sanitize_text(
+            converter_data.capella_element,
+            rendered_jinja,
+            converter_data.errors,
+        )
 
     def setup_env(self, env: jinja2.Environment) -> None:
         """Add the link rendering filter."""
         env.filters["make_href"] = self.__make_href_filter
+        env.filters["render_diagram_with_params"] = (
+            self.__render_diagram_with_params
+        )
         env.globals["insert_diagram"] = self.__insert_diagram
 
     def __make_href_filter(self, obj: object) -> str:
         if (obj := self.check_model_element(obj)) is None:
             return "#"
         return f"hlink://{obj.uuid}"
 
+    def __render_diagram_with_params(
+        self,
+        diagram: m.AbstractDiagram,
+        format: str | None = None,
+        render_params: dict[str, t.Any] | None = None,
+    ) -> t.Any | cdiagram.Diagram:
+        """Render a diagram with the given parameters."""
+        return diagram.render(format, **(render_params or {}))
```

Comment:
This helper transforms calls like this:

    {{ diagram.render("svg", **params) }}
    {{ diagram.render("svg", param1=True, param2=False) }}

into:

    {{ diagram | render_diagram_with_params("svg", params) }}
    {{ diagram | render_diagram_with_params("svg", {"param1": True, "param2": False}) }}

Which is not only much longer, but also changes the interface from `**params` to `params` as a dict.

What's the point of this? When and why would one use this over just calling `diagram.render()` directly?